search/render.go

```diff
@@ -55,16 +55,26 @@ func splitRxTerms(rx string) pgtype.TextArray {
 			continue
 		}
 
-		terms = append(terms, "\\m"+cur)
+		// prefix match terms with the \m "word start" symbol
+		if rx == "" {
+			rx = "\\m" + cur
+		} else {
+			// extra words in between are allowed with .*
+			rx = rx + ".*\\m" + cur
+		}
```

Comment:
Maybe a simpler way would be to change the original `splitRxTerms` func return `[]string` and then have a new `orderedPrefixRx` func call it and join with `.*`

mixing the join with the term separation is a little hard to follow